### PR TITLE
Support verifying SELECT queries with DATE or UNKNOWN columns

### PR DESCRIPTION
**Pull request by caithagoras**
```
== RELEASE NOTES ==

Verifier Changes
* Add support for verifying ``SELECT`` queries that produce ``DATE`` or ``UNKNOWN`` (null) columns.
```

### CODE REVIEW

**Comment by mbasmanova** [comment]
@caithagoras What is UNKNOWN column and what are the queries that return no results?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java [hunk 1]
```diff
@@ -191,7 +205,62 @@ private QualifiedName generateTemporaryTableName(Optional<QualifiedName> origina
...
+                        querySpecification.getWhere(),
+                        querySpecification.getGroupBy(),
+                        querySpecification.getHaving(),
+                        querySpecification.getOrderBy(),
+                        querySpecification.getLimit()),
+                query.getOrderBy(),
+                query.getLimit());
+    }
+
+    private static Optional<TypeSignature> getColumnTypeRewrite(TypeSignature type)
```
Comment (mbasmanova) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java]: This logic is Hive-specific. Does Verifier support only Hive connector?

**Comment by caithagoras** [comment]
@mbasmanova A Presto `unknown` type corresponds to a Hive `void` type, whose value can only be `null`. For example, `SELECT null` produces a column of `unknown` type.

Although being able to verify `unknown` column is trivial (they always matches), a query may have a mixture of `unknown` columns and normal columns, and it's better if we verify those queries than skipping them. There are a number of reasons why people want so select multiple null columns: query might be generated, results may be consumed by a script, and so on.

In additional to those change, we'll also be able to support structured type with `DATE` and `UNKNOWN`, etc.

**Comment by mbasmanova** [comment]
@caithagoras Thanks for explaining. I understand now what is top-level UNKNOWN column, but I'm still puzzled at how list, map, struct may include a field of type UNKNOWN. Would you clarify?

**Comment by caithagoras** [comment]
This is similar with structure type. If array elements are statically `null`, the element type is `unknown`, and the array time is `array(unknown)`, which cannot be written into Hive.
```
explain select array[null, null]
- Output[_col0] => [expr:array(unknown)]

explain select map(array[1], array[null])
- Output[_col0] => [map:map(integer, unknown)]

explain select row(null)
- Output[_col0] => [expr:row(unknown)]
```

**Comment by mbasmanova** [comment]
@caithagoras Thank you so much for clarifying.

**Comment by caithagoras** [comment]
@mbasmanova Regd `Improve JdbcPrestoAction`:

When we run a Presto query using jdbc, `INSERT INTO` query does not have a `ResultSet`, that is, `PrestoStatement.getResultSet` returns `null`, and `getUpdateCount` returns the number of rows inserted. I haven't checked other query types but I think `CREATE TABLE AS` should be similar.

"no result" is not a good wording, as "ignore result" is closer to what is done there.
`JdbcPrestoAction` provides 2 methods, one that takes `ResultSetConverter` and returns a `QueryResult`, and the other does not takes a converter and returns a `QueryStats`. If caller does not provide a converter, the query will be executed, but the result will be ignored if the query is a result-producing query. We do not need results for setup, main, and teardown queries, and we do need results for rewrite, desc, and checksum queries.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java (hunk 1 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java]: This is the assumption so far. Verifier are unaware about what connector is associated with a given catalog, so it won't be able make decisions around that as well.

**Pull request merged**